snapshotserver/main.go

```diff
@@ -153,6 +156,22 @@ func Close() {
 	}
 }
 
+func basicValidationHandler(h httprouter.Handle) httprouter.Handle {
+	var re = regexp.MustCompile(scopeParamValidChars)
```

Comment:
Are we sure that we aren't compiling the regexp on every HTTP request in this case? I know that if we move the "MustCompile" into a global variable than it will only